### ISSUE TITLE
T Container.GetInstance(ExplicitArguments args) fails with exception

### ISSUE THREAD

**Issue by AnotherICe** [structuremap/structuremap#563]
T Container.GetInstance<T>(ExplicitArguments args) fails with exception 
> StructureMap.StructureMapConfigurationException:
> No default instance or named instance 'Default' for requested plugin type
upon following circumstances:
1. container configuration with PluginType registered by .For(Type).Use(Type)
2. GetInstance called with ExplicitArguments

Problem diagnostics:
1. .For(Type).Use(Type) creates a ConfiguredInstance
2.  T GetInstance<T>(ExplicitArguments args) calls Override (in BuildInstanceWithArgs method) method  of IOverridableInstance implementation 
3. This falls to ConfiguredInstance.Override (which not exists) which falls to ConstructorInstance<ConfiguredInstance> override
4. arguments.OnlyNeedsDefaults() && !this.Dependencies.HasAny() returns true and As<ConstructorInstance> is called
But, ConstructorInstance is inherited from ConstructorInstance<ConstructorInstance> and ConfiguredInstance is inherited from ConstructorInstance<ConfiguredInstance> and because of this this.As<ConstructorInstance> call on line 102 returns null. This leads to exception on resolve stage.

Reproducer:

```
public static class Program
	{
		public static void Main()
		{
			var container = new Container(x =>
			{
				x.For(typeof(IMyService)).Use(typeof(MyService));
				//x.For<IMyService>().Use<MyService>(); // NOTE: this one works OK
			});

			var explicitArguments = new ExplicitArguments();

			// NOTE: Throws Unhandled Exception: 
			// StructureMap.StructureMapConfigurationException: No default instance or named instance Default for requested plugin type StructureMap.Reproducer.IMyService
			var myService = container.GetInstance<IMyService>(explicitArguments);
		}
	}

	public interface IMyService
	{
		
	}
	public class MyService : IMyService
	{
		
	}

```

**Comment by jeremydmiller** [comment]
Looking at it now. I've got 3-4 bugs going w/ StructureMap right now all related to the With() functionality and recent changes.

**Comment by jeremydmiller** [comment]
The fix for this one will be in StructureMap 4.4.5 later today
